source/fundamentals/connection/tls.txt

```diff
@@ -0,0 +1,317 @@
+
+Overview
+--------
+
+In this guide, you can learn how to connect to MongoDB instances with the
+`TLS/SSL <https://en.wikipedia.org/wiki/Transport_Layer_Security>`__
+security protocol. To configure your connection to use TLS/SSL, enable
+the TLS/SSL settings in either your connection string or the
+`MongoClientOptions <{+api+}/interfaces/MongoClientOptions.html>`__
+for your client.
+
+.. important:: TLS vs. SSL
+   
+   TLS is the successor protocol to SSL, but the terms are often used
+   interchangeably. Since TLS is an improved version of SSL, this guide
+   describes how to connect to MongoDB with the TLS protocol.
+
+.. _node-enable-tls:
+
+Enable TLS
+----------
+
+You can enable TLS on a connection to your MongoDB instance
+in the following ways:
+
+- Setting the ``tls`` option to ``true`` in your ``MongoClientOptions``
+- Setting the ``tls`` option to ``true`` in your connection string
+
+.. tabs::
+
+   .. tab:: MongoClientOptions
+      :tabid: mongoclientoptions
+
+      To enable TLS on a connection by using an option in a
+      ``MongoClientOptions`` instance, set the property ``tls`` as
+      ``true`` and pass the options to the ``MongoClient`` constructor:
+      
+      .. code-block:: js
+         :emphasize-lines: 2
+
+         var myClientSettings = {
+           tls: true,
+           ...
+         };
+         
+         const client = new MongoClient(uri, myClientSettings);
+
+   .. tab:: Connection String
+      :tabid: connection string
+
+      To enable TLS on a connection by using a connection string option,
+      set the parameter ``tls`` as ``true`` in the connection string URI
+      that you pass to the ``MongoClient`` constructor:
+
+      .. code-block:: js
+         :emphasize-lines: 1
+
+         const uri = "mongodb://<hostname>:<port>?tls=true";
+         const client = new MongoClient(uri, myClientSettings);
+
+.. _node-configure-tls-certificates:
+
+Configure Certificates
+----------------------
+
+Applications that initiate TLS requests require access to
+cryptographic certificates that prove the identity of the application
+itself and other applications that the application
+interacts with. To connect to MongoDB with TLS, your certificates must
+be stored as ``PEM`` files, which are concatenated certificate
+containers.
+
+For production use, your MongoDB deployment should use valid certificates generated and
+signed by the same certificate authority. For testing, you can use self-signed certificates.
+
+Configure TLS Client Options
+----------------------------
+
+In addition to the ``tls`` option, the driver provides client options
+that you can use to configure TLS on your connection.
+
+The following table describes the TLS client options that you can
+specify to reference your certificates and connect to MongoDB with TLS
+enabled. The first column describes Node.js options that you can specify
+when creating a :ref:`SecureContext <node-tls-securecontext>`. The second column describes
```

Comment:
Node.js can natively handle the certificates if you use the Node.js ``tls`` package. Alternatively, you can use the options given by the Node driver. I struggled to explain this because the explanation of secureContext is given in a later section. Do you think there should be a sentence here explaining when/why you should use the node options as opposed to the driver options?

Reply:
[this](https://mongodb.github.io/node-mongodb-native/5.2/interfaces/MongoOptions.html#tls) is the table I was trying to reproduce